cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -1937,6 +1967,25 @@ pub struct MachTrap {
 pub struct MachCallSite {
     /// The offset of the call's return address, *relative to the containing section*.
     pub ret_addr: CodeOffset,
+
+    /// Range in `exception_handlers` corresponding to the exception
+    /// handlers for this callsite.
+    exception_handler_range: Range<u32>,
+}
+
+/// A call site record resulting from a compilation.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(
+    feature = "enable-serde",
+    derive(serde_derive::Serialize, serde_derive::Deserialize)
+)]
+pub struct FinalizedMachCallSite {
```

Comment:
This doesn't need to be a separate type anymore as `MachCallSite` no longer contains `MachLabel`s.

Reply:
Or actually, maybe we should store `MachCallSite` and have `buffer.call_sites()` return owned `FInalizedMachCallSite<'a>` values with the `exception_handler_range` field replaced by `exception_handler: &'a [(PackedOption<ir::ExceptionTag>, CodeOffset)]`?

Reply:
Ah, yes, I like the latter option a lot -- updated, thanks!